policy-controller/k8s/index/src/outbound/index/unmeshed_network.rs

```diff
@@ -0,0 +1,230 @@
+use chrono::{offset::Utc, DateTime};
+use linkerd_policy_controller_k8s_api::policy::Cidr;
+use linkerd_policy_controller_k8s_api::{policy as linkerd_k8s_api, ResourceExt};
+use std::net::IpAddr;
+
+#[derive(Debug, Default)]
+pub(crate) struct Networks(pub Vec<Cidr>);
+
+#[derive(Debug, Default)]
+pub(crate) struct UnmeshedNetwork {
+    pub networks: Networks,
+    pub name: String,
+    pub namespace: String,
+    pub creation_timestamp: Option<DateTime<Utc>>,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+struct MatchedUnmeshedNetwork {
+    matched_cidr: MatchedCidr,
+    name: String,
+    namespace: String,
+    creation_timestamp: Option<DateTime<Utc>>,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+struct MatchedCidr(Cidr);
+
+// === impl MatchedUnmeshedNetwork ===
+
+impl Networks {
+    pub fn find_match(&self, ip: IpAddr) -> Option<Cidr> {
+        let ip: Cidr = ip.into();
+        self.0
+            .iter()
+            .filter(|c| c.contains(&ip))
+            .min_by(|a, b| a.size().cmp(&b.size()))
+            .cloned()
+    }
+}
+
+// === impl UnmeshedNetwork ===
+
+impl From<linkerd_k8s_api::UnmeshedNetwork> for UnmeshedNetwork {
+    fn from(u: linkerd_k8s_api::UnmeshedNetwork) -> Self {
+        let name = u.name_unchecked();
+        let namespace = u
+            .namespace()
+            .expect("UnmeshedNetwork must have a namespace");
+
+        UnmeshedNetwork {
+            name,
+            namespace,
+            networks: Networks(u.spec.networks.clone()),
+            creation_timestamp: u.creation_timestamp().map(|d| d.0),
+        }
+    }
+}
+
+// === impl MatchedUnmeshedNetwork ===
+
+impl From<(Cidr, &UnmeshedNetwork)> for MatchedUnmeshedNetwork {
+    fn from((cidr, unet): (Cidr, &UnmeshedNetwork)) -> Self {
+        Self {
+            name: unet.name.clone(),
+            namespace: unet.namespace.clone(),
+            matched_cidr: MatchedCidr(cidr),
+            creation_timestamp: unet.creation_timestamp,
+        }
+    }
+}
+
+impl std::cmp::PartialOrd for MatchedUnmeshedNetwork {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl std::cmp::Ord for MatchedUnmeshedNetwork {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.matched_cidr
+            .cmp(&other.matched_cidr)
+            .then_with(|| self.creation_timestamp.cmp(&other.creation_timestamp))
+            .then_with(|| self.namespace.cmp(&other.namespace).reverse())
+            .then_with(|| self.name.cmp(&other.name).reverse())
+    }
+}
```

Comment:
what does sorting this mean, why do we need to sort these?

Reply:
i think this ties into my feedback in this thread, regarding `Ord` implementations and business logic https://github.com/linkerd/linkerd2/pull/13150/files#r1790642130

Reply:
Good point. This should be way way more explicit.

Reply:
I have removed Ord impls and relied on `max_by`